source/initial_temperature/anelasticity_temperature.cc

```diff
@@ -0,0 +1,421 @@
+    AnelasticVs2T<dim>::
+    fVs (double x,
+         const double      depth,
+         const double      absolute_Vs,
+         const double      mu0,
+         const double      dmudT,
+         const double      dmudP,
+         const double      viscosity_prefactor,
+         const double      activation_energy,
+         const double      activation_volume,
+         const double      solidus_gradient,
+         const bool      use_original_model) const
+    {
+      return std::abs(yamauchi_takei_Vs(x,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
+                                        ,solidus_gradient,use_original_model)-absolute_Vs);
+    }
+
+    // set up volume change function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fdV (double x, const double bulk_modulus, const double bulk_modulus_pressure_derivative, const double pressure ) const
+    {
+      return std::abs((bulk_modulus*(3./2.)*(std::pow(x,7./3.)-std::pow(x,5./3.))*(1+(((3./4.)*
+                                                                                      (bulk_modulus_pressure_derivative-4))*(std::pow(x,2./3.)-1))))-pressure);
+    }
+
+    // set up initial temperature
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    initial_temperature (const Point<dim> &position) const
+    {
+      // determine depth
+      double depth;
+      depth = this->get_geometry_model().depth(position);
+
+      // declare temperature
+      double temperature;
+
+      // read absolute Vs in from ascii file
+      const double absolute_Vs = Utilities::AsciiDataInitial<dim>::get_data_component(position,0);
+
+      if (depth >= no_perturbation_depth)
+        {
+          // convert absolute Vs into temperature
+          // check if using Yamauchi & Takei 2016 parameterization
+          if (use_yamauchi_takei == true)
+            {
+              // specify Brent algorithm parameters
+              const double a=273;
+              const double b=3273;
+              typedef std::pair<double, double> Result;
+              // create fVs function to use in Brent minimization and calculate temperature
+              auto bfunc = [ &,this] (double x)
+              {
+                return fVs(x, depth, absolute_Vs, mu0, dmudT,dmudP,
+                           viscosity_prefactor,activation_energy,activation_volume,solidus_gradient,use_original_model);
+              };
+              // determine maximum Vs
+              double maximum_Vs;
+              int fail;
+              maximum_Vs=yamauchi_takei_Vs(273.,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
+                                           ,solidus_gradient,use_original_model);
+
+              // set number of fails to zero
+              fail=0;
+              // where absolute Vs exceeds maximum Vs, set temperature to 273 K
+              if (absolute_Vs>maximum_Vs)
```

Comment:
Nice error check! We have a cleaner way for doing this though:
```
AssertThrow(absolute_Vs < maximum_Vs, 
                      ExcMessage("Computed Vs of " + std::to_string(absolute_Vs) + " too fast for sensible temperature"));
```
This will not give you the information about the number of failing points, but how much does that help you anyway? It will only report back the number of failing points in each cell separately. Do you need this information?

Reply:
The number of failed points isn't that relevant, most useful is the warning that some of the Vs values you're reading in are too high to give sensible temperatures. I have now implemented your suggested change as I agree it's much cleaner and does what's needed.